frmts/gtiff/tif_jxl.c

```diff
@@ -369,6 +380,58 @@ JXLPreDecode(TIFF* tif, uint16_t s)
                     _TIFFfree(extra_channel_buffer);
                     return 0;
                 }
+
```

Comment:
I was worried that these extra tests might raise warnings on tifjxl files created before we began supporting more than 4 bands, as in that case the jxl codestream had marked the 4th band as alpha whereas that might not have been the case in the tif's ExtraSample[0]. As it happens this series of validations is not activated for this configuration, as `nFirstExtraChannel < info.num_extra_channels` will be false.
I'd like to add such an "old style" jxl tif to the autotests, how can I write a test to ensure that `TIFFWarningExt` was not called?

Reply:
> how can I write a test to ensure that TIFFWarningExt was not called?

similarly to how I've enhanced the test you added yesterday
```
                    gdal.ErrorReset()
                    got_cs = ds.GetRasterBand(idx + 1).Checksum()
                    assert gdal.GetLastErrorMsg() == ""
```